### PR TITLE
feat(module): add highlight support for nvim-navic

### PR DESCRIPTION
**Pull request by simeonoff**
This PR adds highlight support for the [nvim-navic](https://github.com/SmiteshP/nvim-navic) plugin.

### CODE REVIEW

File: lua/nightfox/group/modules/navic.lua [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+local M = {}
+
+function M.get(spec, config, opts)
+  -- stylua: ignore
+  return {
+    NavicText               = { fg = spec.fg1 },
+    NavicSeparator          = { fg = spec.bg4 },
+
+    NavicIconsFile          = { link = "Directory" },
+    NavicIconsModule        = { link = "TSNamespace" },
```
Comment (EdenEast) [inline lua/nightfox/group/modules/navic.lua:12]: `TS*` highlight groups are the old deprecated format. You should use the new `@` format. See the [treesitter](https://github.com/EdenEast/nightfox.nvim/blob/main/lua/nightfox/group/modules/treesitter.lua) module file for more examples.

As an example `TSNamespace` -> `@namespace`.

**Review by EdenEast — CHANGES_REQUESTED** [review]
Looks good just had one comment in the treesitter highlight groups used.

File: lua/nightfox/group/modules/navic.lua (hunk 1 above)
Comment (simeonoff) [reply lua/nightfox/group/modules/navic.lua]: Updated them, thanks!

**Review by EdenEast — APPROVED** [review]
Looks good LGTM!

**Pull request merged**